### ISSUE TITLE
[CI] TimeSeriesTsidHashCardinalityIT testTimeSeriesNumberOfBuckets failing

### ISSUE THREAD

**Issue by dnhatn** [elastic/elasticsearch#105104]
**Build scan:**
https://gradle-enterprise.elastic.co/s/z5cbaxvf3a4ei/tests/:modules:aggregations:internalClusterTest/org.elasticsearch.aggregations.bucket.TimeSeriesTsidHashCardinalityIT/testTimeSeriesNumberOfBuckets

**Reproduction line:**
```
./gradlew ':modules:aggregations:internalClusterTest' --tests "org.elasticsearch.aggregations.bucket.TimeSeriesTsidHashCardinalityIT.testTimeSeriesNumberOfBuckets" -Dtests.seed=3DF1D7F789B2ACC7 -Dtests.locale=id-ID -Dtests.timezone=Atlantic/South_Georgia -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
[Failure dashboard for `org.elasticsearch.aggregations.bucket.TimeSeriesTsidHashCardinalityIT#testTimeSeriesNumberOfBuckets`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testTimeSeriesNumberOfBuckets'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.aggregations.bucket.TimeSeriesTsidHashCardinalityIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.OutOfMemoryError: Java heap space

  at __randomizedtesting.SeedInfo.seed([3DF1D7F789B2ACC7:5536B5155BD292A2]:0)
  at com.fasterxml.jackson.core.util.BufferRecycler.balloc(BufferRecycler.java:190)
  at com.fasterxml.jackson.core.util.BufferRecycler.allocByteBuffer(BufferRecycler.java:135)
  at com.fasterxml.jackson.core.util.BufferRecycler.allocByteBuffer(BufferRecycler.java:125)
  at com.fasterxml.jackson.core.io.IOContext.allocWriteEncodingBuffer(IOContext.java:249)
  at com.fasterxml.jackson.core.json.UTF8JsonGenerator.<init>(UTF8JsonGenerator.java:130)
  at com.fasterxml.jackson.core.JsonFactory._createUTF8Generator(JsonFactory.java:1916)
  at com.fasterxml.jackson.core.JsonFactory.createGenerator(JsonFactory.java:1369)
  at org.elasticsearch.xcontent.provider.json.JsonXContentImpl.createGenerator(JsonXContentImpl.java:84)
  at org.elasticsearch.xcontent.XContentBuilder.<init>(XContentBuilder.java:262)
  at org.elasticsearch.xcontent.XContentBuilder.<init>(XContentBuilder.java:203)
  at org.elasticsearch.xcontent.XContentBuilder.builder(XContentBuilder.java:59)
  at org.elasticsearch.xcontent.provider.json.JsonXContentImpl.getContentBuilder(JsonXContentImpl.java:37)
  at org.elasticsearch.xcontent.provider.XContentProviderImpl$2.getContentBuilder(XContentProviderImpl.java:48)
  at org.elasticsearch.xcontent.json.JsonXContent.contentBuilder(JsonXContent.java:30)
  at org.elasticsearch.xcontent.XContentFactory.contentBuilder(XContentFactory.java:109)
  at org.elasticsearch.xcontent.XContentFactory.jsonBuilder(XContentFactory.java:33)
  at org.elasticsearch.aggregations.bucket.TimeSeriesTsidHashCardinalityIT.setUp(TimeSeriesTsidHashCardinalityIT.java:136)
  at java.lang.invoke.LambdaForm$DMH/0x00007f57fc288000.invokeVirtual(LambdaForm$DMH:-1)
  at java.lang.invoke.LambdaForm$MH/0x00007f57fc0a4800.invoke(LambdaForm$MH:-1)
  at java.lang.invoke.Invokers$Holder.invokeExact_MT(Invokers$Holder:-1)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:153)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)

```

**Comment by dnhatn** [comment]
Muted this suite in https://github.com/elastic/elasticsearch/commit/40a61abb952e0cb2f3c49124a2af7ad1029b40e1

**Comment by martijnvg** [comment]
I suspect the bulk request is growing out of control here. The bulk request should be flushed earlier. Maybe the number of time series can be reduced as well.

**Comment by martijnvg** [comment]
Should be addressed by #105121
